test/server/admin/stats_request_test_base.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/stats/thread_local_store.h"
+#include "source/server/admin/grouped_stats_request.h"
+#include "source/server/admin/ungrouped_stats_request.h"
+
+#include "test/mocks/event/mocks.h"
+#include "test/mocks/stats/mocks.h"
+#include "test/mocks/thread_local/mocks.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Server {
+
+template <class T> class StatsRequestTestBase : public testing::Test {
+protected:
+  StatsRequestTestBase();
+  ~StatsRequestTestBase() override;
+
+  uint32_t iterateChunks(T& request, bool drain = true, Http::Code expect_code = Http::Code::OK);
+
+  Stats::StatName makeStatName(absl::string_view name) { return pool_.add(name); }
+
+  std::string response(T& request);
+
+  Stats::SymbolTableImpl symbol_table_;
+  Stats::StatNamePool pool_;
+  Stats::AllocatorImpl alloc_;
+  testing::NiceMock<Stats::MockSink> sink_;
+  testing::NiceMock<Event::MockDispatcher> main_thread_dispatcher_;
+  testing::NiceMock<ThreadLocal::MockInstance> tls_;
+  Stats::ThreadLocalStoreImpl store_;
+  Buffer::OwnedImpl response_;
+};
+
+class TestScope : public Stats::IsolatedScopeImpl {
```

Comment:
I missed this before. Why do we need this version of TestScope?

Can't we just use MockStore::createScope()?

Actually come to think of it, can we just use IsolatedStoreImpl directly, or maybe Stats::TestUtil::TestStore and TestScope, without introducing these new wrappers?

Reply:
I think the issue with using existing test infrastructure for scope|store boils down to histograms: the `PrometheusStatsFormatterTest` tests heavily rely on mocking behaviour for histograms and I don't see an easy way to plug that into `IsolatedScopeImpl`.

I also looked a bit at `MockScope|Store` but I don't think they can be used as is for those tests: I might be missing something here, so LMK in case I am wrong :)

Reply:
Let's leave this like you have it but it's a bit of technical debt to have to subclass IsolatedScopeImpl for an individual set of tests. There are so many Scope impls we should be able to re-use one.

Can you add a TODO for me to look at this and try to think of how to clean it after this is merged?